libs/trpc/src/lib/client/trpc-rxjs-proxy.ts

```diff
@@ -0,0 +1,213 @@
+import { OperationContext, OperationLink, TRPCClientError, TRPCClientRuntime } from '@trpc/client';
+import type {
+	AnyMutationProcedure,
+	AnyProcedure,
+	AnyQueryProcedure,
+	AnyRouter,
+	CombinedDataTransformer,
+	DataTransformerOptions,
+	IntersectionError,
+	ProcedureArgs,
+	ProcedureRouterRecord,
+	ProcedureType,
+	inferProcedureInput,
+	inferProcedureOutput,
+} from '@trpc/server';
+import { Observable as TrpcObservable, inferObservableValue, share } from '@trpc/server/observable';
+import { createFlatProxy, createRecursiveProxy, inferTransformedProcedureOutput } from '@trpc/server/shared';
+import { Observable as RxJSObservable } from 'rxjs';
+import { CreateTRPCClientOptions, TRPCRequestOptions, TRPCType, createChain } from './shared-internal';
+
+// Changed to rxjs observable
+type Resolver<TProcedure extends AnyProcedure> = (
+	...args: ProcedureArgs<TProcedure['_def']>
+) => RxJSObservable<inferTransformedProcedureOutput<TProcedure>>;
+
+// Removed subscription and using new type
+type DecorateProcedure<
+	TProcedure extends AnyProcedure,
+	TRouter extends AnyRouter,
+> = TProcedure extends AnyQueryProcedure
+	? {
+			query: Resolver<TProcedure>;
+	  }
+	: TProcedure extends AnyMutationProcedure
+	  ? {
+				mutate: Resolver<TProcedure>;
+	    }
+	  : never;
```

Comment:
Prettier looks off here, but it might just be GitHub's rendering.

Reply:
I'll double check this locally!